classes/local/util/manager.php

```diff
@@ -289,27 +289,45 @@ protected function prepare_mod_files() {
 
         $this->files['lib.php']->add_supported_feature('FEATURE_MOD_INTRO');
 
-        // 'mod/<modname>:addinstance' and 'mod/<modname>:view' capabilities are recommended.
-        if (!$this->should_have('capabilities')) {
-            $this->logger->warning('Capabilities not defined');
+        if ($this->should_have('backup_moodle2')) {
+            $this->prepare_mod_backup_moodle2();
+            $this->files['lib.php']->add_supported_feature('FEATURE_BACKUP_MOODLE2');
+        } else {
+            $this->logger->warning('Backup_moodle2 feature not defined');
         }
 
-        if (!$this->should_have('observers')) {
-            $this->logger->warning('Observers not defined');
+        if ($this->should_have('navigation')) {
+            $this->files['lib.php']->set_attribute('has_navigation');
         }
+    }
 
-        if (!$this->should_have('upgrade')) {
-            $this->logger->warning('Upgrade feature not defined');
-        }
+    /*
+     * Prepares the skeleton files for the 'backup_moodle2' feature for an activity module.
+     */
+    public function prepare_mod_backup_moodle2() {
```

Comment:
According to the [Backup API](https://docs.moodle.org/dev/Backup_API) each plugin type implements the backup feature differently: different file names, different files, different file content.

I chose to have a function like this for each plugin type that implements the backup feature. So there'll be:
- function prepare_mod_backup_moodle2() to prepare the backup files for activity modules
- function prepare_block_backup_moodle2() to prepare the backup file for blocks
- function prepare_themes_backup_moodle2() to prepare the backup for themes
- function prepare_report_backup_moodle2() to prepare the backup for reports. I've found no documentation about this, and no report plugin that implements this feature. I think we'll leave this out for now.